Sources/CheckoutNetworkFakeClient/CheckoutNetworkFakeClient.swift

```diff
@@ -36,6 +37,9 @@ extension CheckoutNetworkFakeClient {
   public func runRequest(with configuration: RequestConfiguration) async throws {
       calledAsyncRequests.append(configuration)
       try await Task.sleep(nanoseconds: 1 * 1_000_000_000) // 1 second
-      return ()
+    guard let error = expectedError else {
```

Comment:
is it worth adding the same for the other async function that returns data so we can get it over with all at once?